sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -579,14 +667,43 @@ public Read withTemplateCompatibility() {
         BATCH
     }
 
+    /**
+     * An enumeration type for the method to be used when reading data from BigQuery.
+     */
+    public enum Method {
+
+      /**
+       * The default behavior if no method is explicitly set. Currently {@link #GCS_EXPORT}.
+       */
+      DEFAULT,
+
+      /**
+       * Export data to Google Cloud Storage in Avro format and read data from that location. This
+       * option can be used to read from existing BigQuery tables and to read the results of
+       * queries, and requires that callers specify a temporary Google Cloud Storage bucket to
```

Comment:
This is a bit misleading: it uses the pipeline's temp directory, and it's not possible to override it currently.

Reply:
I will update the comment.

Reply:
Done.